script/make_release.py

```diff
@@ -0,0 +1,262 @@
+    else:
+        raise ValueError("file {} is not a file or dir.".format(path))
+
+def copytree(src, dst, symlinks=False, ignore=None):
+    '''
+    Copies contents of the `src` directory to `dst` recursively.
+    If `dst` doesn't exist it is created.
+    '''
+    if not os.path.exists(dst):
+        os.mkdir(dst)
+
+    for item in os.listdir(src):
+        s = os.path.join(src, item)
+        d = os.path.join(dst, item)
+        if os.path.isdir(s):
+            shutil.copytree(s, d, symlinks, ignore)
+        else:
+            shutil.copy2(s, d)
+
+def clone_stockfish(destination_dir, branch='master'):
+    '''
+    Clone, with depth 1, the specified `branch` from the `REPO`.
+    The repository is cloned to the `destination_dir`.
+    '''
+    log('Cloning branch {} from repository {} into {}' \
+        .format(branch, REPO, destination_dir))
+
+    subprocess.run( \
+        ['git', 'clone', '--branch', branch, '--depth', '1', REPO, destination_dir], \
+        stdout=STDOUT, stderr=STDERR)
+
+    log('Cloning finished.')
+
+class DummyContextManager():
+    '''
+    Useful for "conditional" `with` statements.
+    '''
+
+    def __enter__(self):
+        return None
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        return False
+
+class TemporaryDirectory():
+    '''
+    Maintains a temporary directory with a unique name.
+    '''
+
+    def __init__(self, prefix=''):
+        self.path = prefix + str(uuid.uuid4())
+        os.mkdir(self.path)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        remove(self.path)
+
+def compile_target(repo_dir, dest_dir, build_type='profile-build', arch='x86-64-modern', prefix=''):
+    '''
+    This function performs the following:
+    1. Copies the contents of `repo_dir` into a temporary directory.
+    2. Runs `make clean` in that temporary directory + ./src/
+    3. Runs `make `build_type`` for the specified `arch`.
+       NOTE: If SDE_PATH is not empty then SDE_PATH is set for all make commands.
+    4. Copies the resulting executable file into `dest_dir` with a given `prefix`
+    '''
+    if prefix != '':
+        prefix = '-' + prefix
+
+    if args.keep_logs:
+        # If the user wants to keep logs a file is created and stdout redirected to it.
+        # Otherwise we use the dummy context manager and stdout is not redirected.
+        logfile_name = os.path.join(dest_dir, 'stockfish{}-{}-{}.log'.format(prefix, build_type, arch))
+        log('{}: Creating log file at {}.'.format(arch, logfile_name))
+
+        logfile = open(logfile_name, 'w')
+        stdout = logfile
+    else:
+        logfile = DummyContextManager()
+        stdout = STDOUT
+
+    with logfile:
+        with TemporaryDirectory('./stockfish-{}-{}-'.format(build_type, arch)) as copy_dir:
+            log('{}: Created temporary directory {}'.format(arch, copy_dir.path))
+
+            src_dir = copy_dir.path + '/src/'
```

Comment:
w/o using `pathlib` better to use:
`src_dir = os.path.join(copy_dir.path, '/src/')`